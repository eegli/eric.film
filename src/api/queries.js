import gql from 'graphql-tag';

export const ALL_BLOGPOSTS_PREVIEW = gql`
  query ALL_BLOGPOSTS_PREVIEW($first: Int!, $skip: Int!) {
    blogposts(first: $first, skip: $skip, orderBy: createdAt_DESC) {
      id
      excerpt
      slug
      title
      type
      previewImage {
        url
      }
    }
    blogpostsConnection {
      aggregate {
        count
      }
    }
  }
`;

export const allBlogPostsVars = {
  skip: 0,
  first: 6,
};

export const SINGLE_BLOGPOST = slug =>
  `query SINGLE_BLOGPOST {
    blogpost(where: {slug: "${slug}"}) {
      id
      excerpt
      content
      title
      slug
    }
  }
  `;

export const ALL_IMGS = gql`
  query ALL_IMGS {
    imgCollections {
      id
      imageType
      collection {
        url
      }
    }
  }
`;
