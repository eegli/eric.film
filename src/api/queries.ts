import { gql } from '@apollo/client';

// Fetch the first 18 posts, skip 0 and sort by latest by default

export const BlogpostsPreview = gql`
  query BlogpostsPreview(
    $first: Int! = 18
    $skip: Int! = 0
    $orderBy: BlogpostOrderByInput! = createdAt_DESC
  ) {
    blogposts(first: $first, skip: $skip, orderBy: $orderBy) {
      id
      excerpt
      slug
      title
      type
      updatedAt
      createdAt
      previewImage {
        url(
          transformation: {
            document: { output: { format: jpg } }
            image: { resize: { height: 400, fit: clip } }
          }
        )
      }
    }
    blogpostsConnection {
      aggregate {
        count
      }
    }
  }
`;

export const Blogpost = gql`
  query Blogpost($slug: String!) {
    blogpost(where: { slug: $slug }) {
      id
      excerpt
      slug
      title
      type
      content
      ytvideo
      updatedAt
      createdAt
      previewImage {
        url(
          transformation: {
            document: { output: { format: jpg } }
            image: { resize: { height: 630, fit: crop, width: 1200 } }
          }
        )
      }
    }
  }
`;

export const ImageCollections = gql`
  query ImageCollections {
    imgCollections {
      id
      imageType
      collection {
        id
        url(transformation: { document: { output: { format: jpg } } })
      }
    }
  }
`;

// Todo node: Pagination
// Get the post before and after a specific blogpost

/* query Blogpost($item: String!, $type:BlogType) {
  nextPost: blogposts(first: 1, after: $item, where: {type: $type}) {
    id
    title
  }
  lastPost: blogposts(last: 1, before: $item, where: {type: $type}) {
    id
    title
  }
}
 */
