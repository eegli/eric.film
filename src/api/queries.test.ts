import { Blogpost, BlogpostsPreview, ImageCollections } from './queries';

describe('Query BlogpostsPreview', () => {
  it('matches snapshot', () => {
    expect(BlogpostsPreview).toMatchSnapshot();
  });
});

describe('Query Blogpost', () => {
  it('matches snapshot', () => {
    expect(Blogpost).toMatchSnapshot();
  });
});

describe('Query ImageCollections', () => {
  it('matches snapshot', () => {
    expect(ImageCollections).toMatchSnapshot();
  });
});
