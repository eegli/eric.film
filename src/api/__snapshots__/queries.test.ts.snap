// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query Blogpost matches snapshot 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "Blogpost",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "where",
                },
                "value": Object {
                  "fields": Array [
                    Object {
                      "kind": "ObjectField",
                      "name": Object {
                        "kind": "Name",
                        "value": "slug",
                      },
                      "value": Object {
                        "kind": "Variable",
                        "name": Object {
                          "kind": "Name",
                          "value": "slug",
                        },
                      },
                    },
                  ],
                  "kind": "ObjectValue",
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "blogpost",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "excerpt",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "slug",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "title",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "type",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "content",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "ytvideo",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "updatedAt",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "createdAt",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "previewImage",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "transformation",
                            },
                            "value": Object {
                              "fields": Array [
                                Object {
                                  "kind": "ObjectField",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "document",
                                  },
                                  "value": Object {
                                    "fields": Array [
                                      Object {
                                        "kind": "ObjectField",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "output",
                                        },
                                        "value": Object {
                                          "fields": Array [
                                            Object {
                                              "kind": "ObjectField",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "format",
                                              },
                                              "value": Object {
                                                "kind": "EnumValue",
                                                "value": "jpg",
                                              },
                                            },
                                          ],
                                          "kind": "ObjectValue",
                                        },
                                      },
                                    ],
                                    "kind": "ObjectValue",
                                  },
                                },
                                Object {
                                  "kind": "ObjectField",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "image",
                                  },
                                  "value": Object {
                                    "fields": Array [
                                      Object {
                                        "kind": "ObjectField",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "resize",
                                        },
                                        "value": Object {
                                          "fields": Array [
                                            Object {
                                              "kind": "ObjectField",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "height",
                                              },
                                              "value": Object {
                                                "kind": "IntValue",
                                                "value": "630",
                                              },
                                            },
                                            Object {
                                              "kind": "ObjectField",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "fit",
                                              },
                                              "value": Object {
                                                "kind": "EnumValue",
                                                "value": "crop",
                                              },
                                            },
                                            Object {
                                              "kind": "ObjectField",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "width",
                                              },
                                              "value": Object {
                                                "kind": "IntValue",
                                                "value": "1200",
                                              },
                                            },
                                          ],
                                          "kind": "ObjectValue",
                                        },
                                      },
                                    ],
                                    "kind": "ObjectValue",
                                  },
                                },
                              ],
                              "kind": "ObjectValue",
                            },
                          },
                        ],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "url",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": undefined,
          "directives": Array [],
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "String",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "slug",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 415,
    "start": 0,
  },
}
`;

exports[`Query BlogpostsPreview matches snapshot 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "BlogpostsPreview",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "first",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "first",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "skip",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "skip",
                  },
                },
              },
              Object {
                "kind": "Argument",
                "name": Object {
                  "kind": "Name",
                  "value": "orderBy",
                },
                "value": Object {
                  "kind": "Variable",
                  "name": Object {
                    "kind": "Name",
                    "value": "orderBy",
                  },
                },
              },
            ],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "blogposts",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "excerpt",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "slug",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "title",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "type",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "updatedAt",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "createdAt",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "previewImage",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "transformation",
                            },
                            "value": Object {
                              "fields": Array [
                                Object {
                                  "kind": "ObjectField",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "document",
                                  },
                                  "value": Object {
                                    "fields": Array [
                                      Object {
                                        "kind": "ObjectField",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "output",
                                        },
                                        "value": Object {
                                          "fields": Array [
                                            Object {
                                              "kind": "ObjectField",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "format",
                                              },
                                              "value": Object {
                                                "kind": "EnumValue",
                                                "value": "jpg",
                                              },
                                            },
                                          ],
                                          "kind": "ObjectValue",
                                        },
                                      },
                                    ],
                                    "kind": "ObjectValue",
                                  },
                                },
                                Object {
                                  "kind": "ObjectField",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "image",
                                  },
                                  "value": Object {
                                    "fields": Array [
                                      Object {
                                        "kind": "ObjectField",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "resize",
                                        },
                                        "value": Object {
                                          "fields": Array [
                                            Object {
                                              "kind": "ObjectField",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "height",
                                              },
                                              "value": Object {
                                                "kind": "IntValue",
                                                "value": "400",
                                              },
                                            },
                                            Object {
                                              "kind": "ObjectField",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "fit",
                                              },
                                              "value": Object {
                                                "kind": "EnumValue",
                                                "value": "clip",
                                              },
                                            },
                                          ],
                                          "kind": "ObjectValue",
                                        },
                                      },
                                    ],
                                    "kind": "ObjectValue",
                                  },
                                },
                              ],
                              "kind": "ObjectValue",
                            },
                          },
                        ],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "url",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
          },
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "blogpostsConnection",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "aggregate",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "count",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [
        Object {
          "defaultValue": Object {
            "kind": "IntValue",
            "value": "18",
          },
          "directives": Array [],
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Int",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "first",
            },
          },
        },
        Object {
          "defaultValue": Object {
            "kind": "IntValue",
            "value": "0",
          },
          "directives": Array [],
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "Int",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "skip",
            },
          },
        },
        Object {
          "defaultValue": Object {
            "kind": "EnumValue",
            "value": "createdAt_DESC",
          },
          "directives": Array [],
          "kind": "VariableDefinition",
          "type": Object {
            "kind": "NonNullType",
            "type": Object {
              "kind": "NamedType",
              "name": Object {
                "kind": "Name",
                "value": "BlogpostOrderByInput",
              },
            },
          },
          "variable": Object {
            "kind": "Variable",
            "name": Object {
              "kind": "Name",
              "value": "orderBy",
            },
          },
        },
      ],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 562,
    "start": 0,
  },
}
`;

exports[`Query ImageCollections matches snapshot 1`] = `
Object {
  "definitions": Array [
    Object {
      "directives": Array [],
      "kind": "OperationDefinition",
      "name": Object {
        "kind": "Name",
        "value": "ImageCollections",
      },
      "operation": "query",
      "selectionSet": Object {
        "kind": "SelectionSet",
        "selections": Array [
          Object {
            "alias": undefined,
            "arguments": Array [],
            "directives": Array [],
            "kind": "Field",
            "name": Object {
              "kind": "Name",
              "value": "imgCollections",
            },
            "selectionSet": Object {
              "kind": "SelectionSet",
              "selections": Array [
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "id",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "imageType",
                  },
                  "selectionSet": undefined,
                },
                Object {
                  "alias": undefined,
                  "arguments": Array [],
                  "directives": Array [],
                  "kind": "Field",
                  "name": Object {
                    "kind": "Name",
                    "value": "collection",
                  },
                  "selectionSet": Object {
                    "kind": "SelectionSet",
                    "selections": Array [
                      Object {
                        "alias": undefined,
                        "arguments": Array [],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "id",
                        },
                        "selectionSet": undefined,
                      },
                      Object {
                        "alias": undefined,
                        "arguments": Array [
                          Object {
                            "kind": "Argument",
                            "name": Object {
                              "kind": "Name",
                              "value": "transformation",
                            },
                            "value": Object {
                              "fields": Array [
                                Object {
                                  "kind": "ObjectField",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "document",
                                  },
                                  "value": Object {
                                    "fields": Array [
                                      Object {
                                        "kind": "ObjectField",
                                        "name": Object {
                                          "kind": "Name",
                                          "value": "output",
                                        },
                                        "value": Object {
                                          "fields": Array [
                                            Object {
                                              "kind": "ObjectField",
                                              "name": Object {
                                                "kind": "Name",
                                                "value": "format",
                                              },
                                              "value": Object {
                                                "kind": "EnumValue",
                                                "value": "jpg",
                                              },
                                            },
                                          ],
                                          "kind": "ObjectValue",
                                        },
                                      },
                                    ],
                                    "kind": "ObjectValue",
                                  },
                                },
                              ],
                              "kind": "ObjectValue",
                            },
                          },
                        ],
                        "directives": Array [],
                        "kind": "Field",
                        "name": Object {
                          "kind": "Name",
                          "value": "url",
                        },
                        "selectionSet": undefined,
                      },
                    ],
                  },
                },
              ],
            },
          },
        ],
      },
      "variableDefinitions": Array [],
    },
  ],
  "kind": "Document",
  "loc": Object {
    "end": 193,
    "start": 0,
  },
}
`;
